pkg/sdk/metrics/metrics.go

```diff
@@ -0,0 +1,76 @@
+package metrics
+
+import (
+	"sync"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	eventsMetricName     = "operator_sdk_events"
+	syncEventsMetricName = "operator_sdk_sync_events"
+	// EventTypeLabel - metric label for event type
+	EventTypeLabel = "type"
+	// EventTypeAdd - addition event label
+	EventTypeAdd = "add"
+	// EventTypeDelete - deletion event label
+	EventTypeDelete = "delete"
+	// EventTypeUpdate - update event label
+	EventTypeUpdate = "update"
+	// SyncResultLabel - metric label for event result
+	SyncResultLabel = "result"
+	// SyncResultSuccess - successful event label
+	SyncResultSuccess = "success"
+	// SyncResultFailure - failed event label
+	SyncResultFailure = "failure"
+)
+
+var (
+	once sync.Once
+)
+
+// Collector - metric collector for all the metrics the sdk will watch
+type Collector struct {
+	Events     *prom.CounterVec
+	SyncEvents *prom.CounterVec
```

Comment:
Can we rename these fields to be more indicative of the metrics that they're tracking:
```Go
EventType
ReconcileResult
```
Or something similar.